### PR TITLE
introducing circleci

### PR DESCRIPTION
**Pull request by cpanato**
Based on the dev meeting we are moving to circleci

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -52,7 +52,9 @@ endif
 govet:
 ifneq ($(HAS_SERVER),)
 	@echo Running govet
-	@$(GO) vet -shadow $$(go list ./server/...) || exit 1
+	$(GO) get -u golang.org/x/tools/go/analysis/passes/shadow/cmd/shadow
+	$(GO) vet $(GOFLAGS) $$(go list ./server/...) || exit 1
+	$(GO) vet -vettool=$(which shadow) $$(go list ./server/...) || exit 1
```
Comment (hanzei) [inline Makefile]: @lieut-data provided a PR to change this on the server. (https://github.com/mattermost/mattermost-server/pull/10389) Should this also be changed here?

Comment (lieut-data) [reply Makefile]: Yes! I found that `$(which shadow)` doesn't actually find `shadow` at all to my dismay (at least in the context of the `Makefile` -- bash works fine). So I've hard-coded the path, and improved other things in the linked PR.

I'm also debating the `-u` in the `go get` -- it makes the process 1s slower than it feels like it ought to be. Thoughts about not forcing the upgrade each time?

Comment (hanzei) [reply Makefile]: Let's continue this discussion on https://github.com/mattermost/mattermost-server/pull/10389/

File: README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Sample Plugin [![Build Status](https://travis-ci.org/mattermost/mattermost-plugin-sample.svg?branch=master)](https://travis-ci.org/mattermost/mattermost-plugin-sample)
+# Sample Plugin [![CircleCI](https://circleci.com/gh/mattermost/mattermost-plugin-sample.svg?style=svg)](https://circleci.com/gh/mattermost/mattermost-plugin-sample)
```
Comment (hanzei) [inline README.md]: Nit: What do you think about using https://shields.io/ instead for the unified style of the badge?

Comment (cpanato) [reply README.md]: we can use nothing against

File: Makefile (hunk 1 above)
Comment (hanzei) [reply Makefile]: @cpanato Please update to 

```suggestion
	$(GO) vet -vettool=$(GOPATH)/bin/shadow $$(go list ./server/...) || exit 1
```

as discussed in https://github.com/mattermost/mattermost-server/pull/10389

**Comment by cpanato** [comment]
@hanzei PTAL

**Pull request merged**